server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/rfc/model/Id.scala

```diff
@@ -0,0 +1,28 @@
+/** **************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ * *
+ * http://www.apache.org/licenses/LICENSE-2.0                 *
+ * *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+
+package scala.org.apache.james.jmap.rfc.model
+
+final case class Id(value: String) {
+  require(Option(value).isDefined, "value cannot be null")
```

Comment:
But i think that in scala we should be able to expect that a value is not null.
And check for the nullity in by using `Option.apply` when interacting with potentially null value from Java.
And when deserializing JSON it should be checked in the Reader which is the behavior of the macro. And using and Option in your case class if you want to explicitly accept null value.

Reply:
@remk I dont get your point. Here the type is nullable java.lang.String, I couldn't find a non-null String type in Scala. What should I do?

Reply:
For me in scala you should expect that a value is not null. Either it is not null or it is an Option.
In scala you can set every variable to null but if you do that and have to check for that in your scala code you will lose your sanity.
It should be done only when interacting with java code and you should wrap the potentially null value into an Option. And not throw exception in your constructors.

Reply:
If re-write my constructor into:
```
final case class Id(value: Option[String])
```

and do not check anything inside, does it make sense in case of an empty Option? I don't think so! an Id need to have a value, right?

Reply:
> If re-write my constructor into:
> 
> ```
> final case class Id(value: Option[String])
> ```
> 
> and do not check anything inside, does it make sense in case of an empty Option? I don't think so! an Id need to have a value, right?

Yes, an Id with an empty value should not exist. 
you should keep the definition ``` 
final case class Id(value: String) 
```

if what you are afraid is that play json will create an id with a null value, don't worry. It will not validate that.

Reply:
`Id` is a generic POJO, not only Play using that. I would rather keep the validations, but they should conform scala best practies :) The problem is, I don't really know how to do it with what you suggested like Using, Try...